src/haxfred.js

```diff
@@ -95,9 +95,37 @@ class Haxfred {
       throw new Error('You have attempted to use a module that does not export a function')
     }
 
+    if (module.requires) this._checkRequiredModules(module.requires)
+
     module(this)
   }
 
+  _checkRequiredModules (requires) {
+    let { adapters = [], components = [], config = [] } = requires
+
+    adapters.forEach(function (adapterName) {
+      this._checkModuleExists('adapter', adapterName)
+    }.bind(this))
+
+    components.forEach(function (componentName) {
+      this._checkModuleExists('component', componentName)
+    }.bind(this))
+
+    config.forEach(function (configPath) {
+      let hasRequiredConfig = get(this.config, configPath)
+
+      if (!hasRequiredConfig) throw new Error(`A module requires the config "${configPath}", but it was not found`)
```

Comment:
Doesn't this require the config value to be truthy? What if we want the param to be defined but it can be `false` etc?